File: atcoder/abc147B_palindrome_philia.nim
# unihernandez22
# https://atcoder.jp/contests/abc147/tasks/abc147_b
# string manipulation

import strutils;

let s = stdin.readLine.strip;

var ans = 0
for i in 0 ..< s.len div 2:
  if s[i] != s[^(i+1)]: inc ans

ans.echo

File: atcoder/abc147B_palindrome_philia.cpp
// unihernandez22
// https://atcoder.jp/contests/abc147/tasks/abc147_b
// implementation

#include<iostream>
#include<string.h>

using namespace std;

int main() {
	string s;
	int ans = 0;
	cin >> s;
	for(int i = 0; i < s.size()/2; i++) {
		if (s[i] != s[s.size()-i-1])
			ans++;
	}
	cout << ans << endl;
	return 0;
}
